src/main/java/com/nvidia/spark/rapids/jni/Hash.java

```diff
@@ -85,7 +85,26 @@ public static ColumnVector xxhash64(ColumnView columns[]) {
     return xxhash64(DEFAULT_XXHASH64_SEED, columns);
   }
 
+  public static ColumnVector hiveHash(ColumnView columns[]) {
+    if (columns.length < 1) {
+      throw new IllegalArgumentException("Hive hashing requires at least 1 column of input");
+    }
```

Comment:
Is this a valid assert? And matched with Spark? I thought that we can call even with 0 column?

Reply:
Spark allows a table to be unpartitioned, yes.  But I don't think it's valid to call the hive hash without specifying at least one input column.

Reply:
To clarify:  Spark shouldn't be invoking the hasher, in the case where no partition/bucketing column is specified.

Reply:
Yeah, Spark ensure there is at least one column when calling into hiveHash.
There are similar assertions for murmur3 and xxhash64.